server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -101,9 +101,10 @@ public IngestService(ClusterService clusterService, ThreadPool threadPool,
                 threadPool.getThreadContext(), threadPool::relativeTimeInMillis,
                 (delay, command) -> threadPool.schedule(
                     command, TimeValue.timeValueMillis(delay), ThreadPool.Names.GENERIC
-                ), this, client
+                ), this, client, threadPool.generic()::execute
```

Comment:
Some tests do need to be modified as they don't expect `ThreadPool#generic()` to be invoked.